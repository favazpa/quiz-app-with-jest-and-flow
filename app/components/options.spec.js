// @flow
import React from 'react';
import {Component} from 'react';
import Options from './options';
import ShallowRenderer from 'react-test-renderer/shallow';
import {findWithType, findAllWithType} from 'react-shallow-testutils';

let renderer = new ShallowRenderer();

describe('src/component/options', () => {
  const renderButton = () => {
    renderer.render(<Options />);
    return renderer.getRenderOutput();
  };

  test('should render correctly', () => {
    const result = renderButton();
    expect(result).toMatchSnapshot();
  });

  test('should press a  button', () => {

    const result = renderButton();

    const OptionBtn = findWithType(result, Options);
    // console.log('OptionButton',OptionButton)
    OptionBtn.props.onPress();

    console.log('result', result);

  });

});