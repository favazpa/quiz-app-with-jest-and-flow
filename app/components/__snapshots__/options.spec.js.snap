// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/options should render correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#d8e5eb",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 60,
        "marginHorizontal": 20,
        "marginVertical": 5,
        "paddingHorizontal": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "orange",
          "borderRadius": 25,
          "height": 45,
          "justifyContent": "center",
          "width": 45,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
      NaN
    </Text>
  </View>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "700",
      }
    }
  />
</TouchableOpacity>
`;
